BaseTools/Conf/tools_def.template.ms

```diff
@@ -32,52 +32,51 @@
 # 1.17 - Asl compilers should be on the path
 # 1.18 - Remove VS2013 support
 # 1.19 - RC.exe from Windows 10 Kit
+# 1.20 - Drop VS 2015, Add VS2019, use latest Windows kits, remove /Gm which is deprecated
 
-#!VERSION=1.19
+#!VERSION=1.20
 
 IDENTIFIER = Default TOOL_CHAIN_CONF
 
-# 
-# VS2015
-DEFINE VS2015x86_BIN    = ENV(VS140COMNTOOLS)\..\..\Vc\Bin
-#//C:\Program Files (x86)\Microsoft Visual Studio 14.0\Vc\bin
-DEFINE VS2015x86_DLL    = ENV(VS140COMNTOOLS)\..\IDE;DEF(VS2015x86_BIN)
-#//C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE;DEF(VS2015x86_BIN)
-DEFINE VS2015x86_BINX64 = DEF(VS2015x86_BIN)\x86_amd64
-
 #
 # VS2017 - New model for VS2017 where there is potential for many versions of the tools.
 # If a specific version is required then the user must set both env variables:
 ## VS150INSTALLPATH:  base install path on system to VC install dir.  Here you will find the Common7 folder, VC folder, etc
 ## VS150TOOLVER:      version number for the VC compiler tools
 ## VS2017x86_DLL - x86 tools dir must come first before common7/ide folder.
 
+#
+# VS2019 - Follow VS2019 where there is potential for many versions of the tools.
+# If a specific version is required then the user must set both env variables:
+## VS160INSTALLPATH:  base install path on system to VC install dir.  Here you will find the Common7 folder, VC folder, etc
+## VS160TOOLVER:      version number for the VC compiler tools
+## VS2019x86_DLL - x86 tools dir must come first before common7/ide folder.
+
 DEFINE VS2017x86_BIN     = ENV(VS150INSTALLPATH)\VC\Tools\MSVC\ENV(VS150TOOLVER)\bin\HostX86\x86
 DEFINE VS2017x86_DLL     = DEF(VS2017x86_BIN);ENV(VS150INSTALLPATH)\Common7\IDE
 DEFINE VS2017x86_BINX64  = ENV(VS150INSTALLPATH)\VC\Tools\MSVC\ENV(VS150TOOLVER)\bin\HostX86\x64
 DEFINE VS2017x86_BINARM  = ENV(VS150INSTALLPATH)\VC\Tools\MSVC\ENV(VS150TOOLVER)\bin\HostX86\arm
 DEFINE VS2017x86_BINA64  = ENV(VS150INSTALLPATH)\VC\Tools\MSVC\ENV(VS150TOOLVER)\bin\HostX86\arm64
 
-DEFINE WINSDK_VERSION   = 10.0.17134.0
+DEFINE VS2019x86_BIN     = ENV(VS160INSTALLPATH)\VC\Tools\MSVC\ENV(VS160TOOLVER)\bin\HostX86\x86
+DEFINE VS2019x86_DLL     = DEF(VS2019x86_BIN);ENV(VS160INSTALLPATH)\Common7\IDE
+DEFINE VS2019x86_BINX64  = ENV(VS160INSTALLPATH)\VC\Tools\MSVC\ENV(VS160TOOLVER)\bin\HostX86\x64
+DEFINE VS2019x86_BINARM  = ENV(VS160INSTALLPATH)\VC\Tools\MSVC\ENV(VS160TOOLVER)\bin\HostX86\arm
+DEFINE VS2019x86_BINA64  = ENV(VS160INSTALLPATH)\VC\Tools\MSVC\ENV(VS160TOOLVER)\bin\HostX86\arm64
+
+
+DEFINE WINSDK_VERSION   = 10.0.17763.0
 DEFINE WINSDK_BIN       = c:\Program Files (x86)\Windows Kits\10\bin\DEF(WINSDK_VERSION)\x64
 DEFINE WINSDKx86_BIN    = c:\Program Files (x86)\Windows Kits\10\bin\DEF(WINSDK_VERSION)\x86
 
 DEFINE WIN_IASL_BIN            = iasl.exe
-DEFINE WIN_ASL_BIN             = asl.exe
+DEFINE WIN_MS_ASL_BIN          = asl.exe
```

Comment:
I don't see this used anywhere. Is it needed?

Reply:
It isn't used in this file but this file is the basis for private platform versions which might choose to use microsoft's asl.exe tool.  Same could be said for MS_ASL_OUTFLAGS.  Not sure i want to remove it at this time because of that.